### ISSUE TITLE
NET-1643 Fix S6802 FN: Lambdas in loop without a direct block are not reported

### ISSUE THREAD

**Issue by sebastien-marichal** [SonarSource/sonar-dotnet#8394]
Lambdas inside a loop without a direct block as a child are not reported by S6802:

```razor
@foreach (var item in Buttons.Where(x => x.Id == "idToFind"))
    @if (item.Id == "idToFind")
    {
        <button @onclick="(e) => Reset(e)">Reset #3</button> @* False Negative *@
    }
```

This is due to the implementation checking if the lambda is inside a direct descendant block of the loop:

```cs
private static bool IsWithinLoopBody(SyntaxNode node) =>
    node.AncestorsAndSelf().Any(x => x is BlockSyntax && x.Parent is ForStatementSyntax or ForEachStatementSyntax or WhileStatementSyntax or DoStatementSyntax);
```

Repros have been added in #8387.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1643](https://sonarsource.atlassian.net/browse/NET-1643)

[NET-1643]: https://sonarsource.atlassian.net/browse/NET-1643?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
